web/modules/custom_weather_module/src/Plugin/Block/CustomWeatherBlock.php

```diff
@@ -126,25 +128,23 @@ private function userWeather() {
     $cacheId = 'weatherCache';
     // @todo use dependency injection
     // phpcs:ignore
-    if ((\Drupal::cache()->get($cacheId))
-      // phpcs:ignore
-      && (\Drupal::cache()->get($cacheId)->data['location'] === $this->userLocation())
-      // phpcs:ignore
-      && (\Drupal::cache()->get($cacheId)->data['conditions'])) {
-      // @todo use dependency injection
-      // phpcs:ignore
-      return \Drupal::cache()->get($cacheId)->data;
+    $cache = \Drupal::cache()->get($cacheId);
+    $weatherData = $this->getWeather();
+    if ($cache
+      && ($cache->data['location'] === $this->userLocation())
```

Comment:
try to use cache tags instead
https://www.drupal.org/docs/drupal-apis/cache-api/cache-tags